File: 3. Arrays and String/167. Two Sum II - Input Array Is Sorted/two_sum.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        idx1 = 0
        idx2 = len(numbers) - 1
        indices = []

        while idx1 < idx2:
            sum = numbers[idx1] + numbers[idx2]
            if sum > target:
                idx2 -= 1
            elif sum < target:
                idx1 += 1
            else:
                indices.append(idx1 + 1)
                indices.append(idx2 + 1)
                return indices

        return indices

File: 3. Arrays and String/167. Two Sum II - Input Array Is Sorted/two_sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &numbers, int target)
    {
        int idx1 = 0;
        int idx2 = numbers.size() - 1;
        vector<int> indices;

        while (idx1 < idx2)
        {
            int sum = numbers[idx1] + numbers[idx2];
            if (sum > target)
                idx2--;
            else if (sum < target)
                idx1++;
            else
            {
                indices.push_back(idx1 + 1);
                indices.push_back(idx2 + 1);
                return indices;
            }
        }
        return indices;
    }
};